### PR TITLE
Add the full-stack code challenge documentation

### PR DESCRIPTION
**Pull request by nicolasiensen**
Jira ticket: RSD-605
---
- [x] Remove the PEP8 sentence from the reviewing section

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,107 @@
...
+## Reviewing
+
+The following description will give you an understanding of how we review the code challenge. What matters to us is to learn how you write code and what you consider as clean code. For us it is more important to have an understandable project than a complex algorithm.
+
+The criteria that we are looking for are the following:
+
+- Documentation: Is the project and the code properly documented?
+- Correctness: Is the task solved? If there is anything missing, is the reason why it is missing documented?
+- Technology: Which libraries or approaches are used? Do they make sense for the task?
+- Code quality: Is the code understandable and maintainable? Does the project make use of Object Oriented Programming? Does the code follow the PEP8 pattern? Is the projected linted?
```
Comment (adamlukemooney) [inline README.md]: Isn't PEP8 specific to Python? Might be confusing for candidates implementing in Ruby/Node

File: README.md [hunk 2]
```diff
@@ -1 +1,107 @@
...
+You can look at the Google Maps, Mapbox and Leaflet libraries as options to help you with building a map visualization. Try to consider how easy your visualization is to understand. Some concepts which can help to improve the visualization:
+
+* Group clusters of vehicles together
+* Modify your visualization depending on zoom level
+
+We expect your visualization to be accessible via a web browser. It only needs to be tested against the most recent version of Chrome.
+
+## Additional business requirements
+
+If a vehicle has exited the "city boundaries", the system should disregard any GPS emissions from that vehicle. The city boundary is currently assumed to be a 50km radius around door2door's office which is located at latitude 52.53 and longitude 13.403.
```
Comment (adamlukemooney) [inline README.md:72]: We should make sure the simulator emits some GPS coordinates beyond the boundaries (not a blocker on this PR, just a comment).

**Review by adamlukemooney — CHANGES_REQUESTED** [review]
Nice job. One minor point below.

File: README.md [hunk 3]
```diff
@@ -1 +1,107 @@
...
+
+All valid position data received must be stored somewhere for future analysis and consumption by the door2door data science team.
+
+## Technical assumptions
+
+* The API is expected to be implemented using one of the following languages: Ruby, Python or Javascript.
+* You are free to make use of any frontend framework or library you please (if any), but you should justify your choice.
```
Comment (yonelacort) [inline README.md:79]: I like the inclusion of `justify your choice`. We should probably also ask for justification for all technology choices (database, API programming language and framework, etc.)

File: README.md [hunk 4]
```diff
@@ -1 +1,107 @@
...
+
+## Optional extras
```
Comment (yonelacort) [inline README.md:81]: Good that these 2 points are added as extras to not scare potential candidates!

File: README.md [hunk 5]
```diff
@@ -1 +1,107 @@
...
+
+## Reviewing
+
+The following description will give you an understanding of how we review the code challenge. What matters to us is to learn how you write code and what you consider as clean code. For us it is more important to have an understandable project than a complex algorithm.
+
+The criteria that we are looking for are the following:
+
+- Documentation: Is the project and the code properly documented?
+- Correctness: Is the task solved? If there is anything missing, is the reason why it is missing documented?
+- Technology: Which libraries or approaches are used? Do they make sense for the task?
```
Comment (yonelacort) [inline README.md:101]: I'd add here _Justify why you've decided to use those technologies to solve the code challenge_

**Comment by nicolasiensen** [comment]
Since we don't have major issues with this pull request, I'm going to merge the new README file, and if you have any changes to suggest you can create a new pull request to apply them, ok?

**Pull request merged**